src/pybind/mgr/dashboard/HACKING.rst

```diff
@@ -3,6 +3,142 @@ Ceph Dashboard Developer Documentation
+  Docker development environment and compiled Ceph there and you now want to
+  switch to your host system, you will also need to recompile Ceph (or
+  compile Ceph using another separate repository).
+
+Development environment on your host system
+...........................................
+
+- No need to learn or have experience with Docker, jump in right away
+- Limited amount of scripts to support automation (like Ceph compilation)
+- No pre-configured easy-to-start services (Prometheus, Grafana, etc)
+- Limited amount of host operating systems supported, depending on which
+  Ceph version is supposed to be used
+
+Development environment using Docker
+....................................
+
+- Some overhead in learning Docker if you are not used to it yet.
+
+- Both Docker projects provide you with scripts that help you getting started
+  and automate recurring tasks.
+
+- Both Docker environments come with partly pre-configured external services
+  which can be used to attach to or complement Ceph Dashboard features, like
+
+  - Prometheus
+  - Grafana
+  - Node-Exporter
+  - Shibboleth
+  - HAProxy
+
+Many of the Ceph Dashboard developers are SUSE employees. We therefore recommend
+you to use `ceph-dev-docker`, which is based on SUSE Tumbleweed. 
+
+vstart on your host system
+~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+The vstart script is usually called from your `build/` directory like so:
+
+.. code::
+
+  ../src/vstart.sh -n -d
+
+In this case `-n` ensures that a new vstart cluster is created (and not the
+previously created on re-used). `-d` enables debug messages in log files.
+There are several more options to chose from. You can get a list using the
+`--help` argument.
+
+Once vstart is running, you will want to check if the `dashboard` manager
+module is running. This can be done by calling `./bin/ceph mgr services` to
+list all the URLs of enabled services. Only URLs of services which are
+available over HTTP(S) will be listed there. Ceph Dashboard is one of these
+services. It should look similar to the following output:
+
+.. code::
+
+  $ ./bin/ceph mgr services
+  {
+      "dashboard": "https://home:41931/",
+      "restful": "https://home:42931/"
+  }
+
+This also reveals the port on which Ceph Dashboard is running on. By default,
+this environment uses a randomly chosen port for Ceph Dashboard and you need
+to use this command to find out which one it has become.
+
+Docker
+~~~~~~
+
+Docker development environments usually ship with a lot of useful scripts.
+`ceph-dev-docker` for instance contains a file called `start-ceph.sh`, which
+cleans up log files, always starts a Rados Gateway service, sets some Ceph
+Dashboard configuration options and automatically runs a frontend proxy, all
+before or after starting up your vstart cluster.
+
+Instructions on how to use those environments are contained in their
+respective repository README files.
+
+`ceph-dev-docker` is mainly used by SUSE employees. `ceph-dev` is the primary
```

Comment:
I would put this whole section right after mentioning ceph-dev and ceph-dev-docker for the first time (line 43-44).
And still +1 for leaving out the "mainly used by [...]"-part and instead only pointing out which OS is supported in which environment. :)

Reply:
I agree, please leave out the company names here and just mention both environments and how they differ from a technology POV.

Reply:
And FWIW, it's "Red Hat", not "RedHat" ;)

Reply:
> And still +1 for leaving out the "mainly used by [...]"-part and instead only pointing out which OS is supported in which environment. :)

It seems I've had that in the doc twice and only removed it once. Regarding the supported operating systems, I'd appreciate some help. For `ceph-dev-docker` it's clear to me, but for `ceph-dev` I'm unsure about the supported operating systems for different Ceph releases.

What I have so far is:

```
          ceph-dev-docker         ceph-dev
Mimic     openSUSE Leap 15        
Nautilus  openSUSE Leap 15
Octopus   openSUSE Leap 15.2
Master    openSUSE Tumbleweed
```

I also know that `ceph-dev` generally supports 
- Fedora
- CentOS 7 / RHEL 7

But not in which constellation. When we already note which operating systems are supported, I'd prefer to have it described in that detail. Can someone fill in those blanks for me? Possibly you, @epuertat ?